src/backend/egl/display.rs

```diff
@@ -212,6 +201,100 @@ impl EGLDisplay {
         })
     }
 
+    /// Create a new [`EGLDisplay`] from an already initialized EGLDisplay and EGLConfig handle
+    pub unsafe fn from_raw<L>(
+        display: *const c_void,
+        config: *const c_void,
+        logger: L,
+    ) -> Result<EGLDisplay, Error>
+    where
+        L: Into<Option<::slog::Logger>>,
+    {
+        let log = crate::slog_or_fallback(logger.into()).new(o!("smithay_module" => "backend_egl"));
+
+        let dp_extensions = ffi::make_sure_egl_is_loaded()?;
```

Comment:
Hmm, this brings up a good question. How can we assert an external context and smithay use the exact same instance of egl?

Reply:
If you're putting the context into smithay wouldn't that be assured? or do you want a function that can tell you if an egl context is the same as the one smithay is using?

Reply:
This is probably something @Drakulix would know about.

Reply:
We can't. It might be that the original library dynamically loaded a different `libEGL`.
There is not really something we can do about it, so we should just add another bullet point to the safety docs.

Reply:
I didn't see this, but I'll add it in! Should it be something like

- Must use the same EGL library smithay does

be good? or what should I put to be more specific?

Reply:
> Must use the same EGL library smithay does

Then we will get questions about "what library does smithay use". :sweat_smile: 
So I would go with something like:

> Not using the system default EGL library (`dlopen("libEGL.so")`) as loaded by smithay might cause undefined behavior